app/src/main/java/com/cornellappdev/transit/ui/viewmodels/FavoritesViewModel.kt

```diff
@@ -58,10 +51,23 @@ class FavoritesViewModel @Inject constructor(
 
             is ApiResponse.Success -> {
                 apiResponse.data.filter { stop ->
-                    favorites[stop.name] == true
+                    favorites.contains(stop.name)
                 }
             }
         }
     }.stateIn(scope, SharingStarted.Eagerly, emptyList())
 
+
+    /**
+     * Function to remove a stop from favorites
+     */
+    suspend fun removeFavorite(stop: String?) {
+        if (stop != null) {
+            val currentFavorites = favoritesFlow.value.toMutableSet()
+            currentFavorites.remove(stop)
+            userPreferenceRepository.setFavorites(currentFavorites.toSet())
+
+        }
+    }
```

Comment:
Hmmm I do like the idea of this function to allow us to remove favorites in the UI! However, it's kinda annoying to launch coroutines all over the place, especially in the UI. We can leverage `viewModelScope` to launch a coroutine with `.launch`. Then, this function no longer has to be `suspend`.